invisible_cities/reco/krmap_functions_test.py

```diff
@@ -104,3 +106,133 @@ def test_transform_parameters(a, b):
+                'pval', 'in_active', 'has_min_counts', 'fit_success', 'valid', 'R', 'X', 'Y']
+
+    assert all(element in columns for element in df.columns.values)
+    assert df.bin.nunique() == n_bins_x*n_bins_y
+
+
+@mark.parametrize('bins_x bins_y'.split(),
+                  ((np.linspace(-100, +100,  5), np.linspace(-100, +100, 10)),
+                  ( np.linspace(-400, +400, 10), np.linspace(-400, +400, 10)),
+                  ))
+def test_get_bin_counts_and_event_bin_id(KrMC_kdst, bins_x, bins_y):
+
+    kr_df  = KrMC_kdst[0].read
+    inside = in_range(kr_df.X, min(bins_x), max(bins_x)) & in_range(kr_df.Y, min(bins_y), max(bins_y))
+
+    counts, bin_labels = krf.get_bin_counts_and_event_bin_id(dst = kr_df, bins_x = bins_x, bins_y = bins_y)
+
+    assert counts.sum()    == kr_df[inside].event.nunique()
+    assert len(bin_labels) == kr_df.event.nunique()
+
+@mark.parametrize('n', (0, 8, 50))
+def test_valid_bin_counter_all_in_active(n):
+    inside = np.ones(50, dtype = bool)
+    valid  = inside.copy()
+    valid[:n//2]  = False
+    valid[len(valid)-n//2:] = False
+    krmap = pd.DataFrame(dict(in_active = inside,
+                              valid     = valid))
+
+    assert krf.valid_bin_counter(krmap) == ((50-n) / 50)
+
+
+def test_valid_bin_counter_valid_subset_of_active():
+    inside = np.ones(50, dtype = bool)
+    inside[:3]  = False
+    inside[-3:] = False
+    valid  = inside.copy()
+    valid[:4]  = False
+    valid[-4:] = False
+    krmap = pd.DataFrame(dict(in_active = inside,
+                              valid     = valid))
+
+    assert krf.valid_bin_counter(krmap) == ((50-6-2) / (50-6))
+
+@mark.parametrize('ni', (0, 4, 25))
+@mark.parametrize('nv', (0, 3, 25))
+def test_valid_bin_counter_valid_outside_active(ni, nv):
+    inside = np.ones(50, dtype = bool)
+    inside[len(inside)-ni:] = False
+    valid  = np.ones(50, dtype = bool)
+    valid[:nv]   = False
+    krmap = pd.DataFrame(dict(in_active = inside,
+                              valid     = valid))
+
+    assert krf.valid_bin_counter(krmap) == ((50-ni-nv) / (50-ni))
+
+@mark.parametrize('n_bins rmax validity_parameter'.split(),
+                  ((( 5, 5), 200,  1.0),
+                   (( 3, 3), 100,  0.2)))
+def test_valid_bin_counter_warning(n_bins, rmax, validity_parameter):
+    counts = np.array(range(n_bins[0]*n_bins[1]))
+    krmap  = krf.create_df_kr_map(bins_x = np.linspace(-rmax, +rmax, n_bins[0]+1),
+                                  bins_y = np.linspace(-rmax, +rmax, n_bins[1]+1),
+                                  counts = counts,
+                                  n_min  = 0,
+                                  r_max  = np.nextafter(np.sqrt(2)*rmax, np.inf))
+
+    krmap.valid.iloc[0 : 9] = True
+
+    if validity_parameter == 1:
+        with warns(UserWarning, match = "inner bins are not valid."):
+            krf.valid_bin_counter(map_df = krmap, validity_parameter = validity_parameter)
```

Comment:
for `validity_parameter = 0.2`, you are not checking anything. Check this: https://stackoverflow.com/a/45671804